compiler/parser/chapel.lex

```diff
@@ -89,21 +89,24 @@ letter           [_a-zA-Z]
 
 ident            {letter}({letter}|{digit}|"$")*
 
-binaryLiteral    0[bB]{bit}+
-octalLiteral     0[oO]{octDigit}+
-hexLiteral       0[xX]{hexDigit}+
-intLiteral       {digit}+|{binaryLiteral}|{octalLiteral}|{hexLiteral}
-
-exponent         [Ee][\+\-]?{digit}+
-floatLiteral1    {digit}*"."{digit}+({exponent})?
-floatLiteral2    {digit}+"."{exponent}
-floatLiteral3    {digit}+{exponent}
+binaryLiteral    0[bB]{bit}(_|{bit})*
+octalLiteral     0[oO]{octDigit}(_|{octDigit})*
+decimalLiteral   {digit}(_|{digit})*
+hexLiteral       0[xX]{hexDigit}(_|{hexDigit})*
+intLiteral       {binaryLiteral}|{octalLiteral}|{decimalLiteral}|{hexLiteral}
+
+digitsOrSeps     {digit}(_|{digit})*
+exponent         [Ee][\+\-]?{digitsOrSeps}
+floatLiteral1    ({digit}*|{digitsOrSeps})"."{digitsOrSeps}({exponent})?
```

Comment:
I think this `({digit}*|{digitsOrSeps})` would be more reasonably expressed as `(|{digitsOrSeps})` but if that doesn't work for some reason what you have is OK.

Reply:
Now that you've suggested that this line could be improved, I think even better would be `{digitsOrSeps}?`.  I see that there are a couple unnecessary (but not outright harmful either) `+` operators in there too that I can clean up. I'll go ahead and do that and do another round of testing.

Reply:
Another round of testing has passed, so I think this is all set.  Thanks @mppf!